template-compose/data/src/main/java/co/nimblehq/template/compose/data/repository/UserPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,54 @@
+package co.nimblehq.template.compose.data.repository
+
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import co.nimblehq.template.compose.domain.model.UserPreferences
+import co.nimblehq.template.compose.domain.repository.UserPreferencesRepository
+import kotlinx.coroutines.flow.*
+import java.io.IOException
+import javax.inject.Inject
+
+class UserPreferencesRepositoryImpl @Inject constructor(private val dataStore: DataStore<Preferences>) :
+    UserPreferencesRepository {
+
+    private val TAG: String = "UserPreferencesRepo"
```

Comment:
Should we make it as `const val` and put it on the top level of class 🤔 
```suggestion
    private const val TAG: String = "UserPreferencesRepo"
```

Reply:
@Wadeewee I don't think we should. As It's just a TAG string and we use it only inside this class. 🤔  To raise it as a constant will waste memory. Can you share the benefit to do that for me?

Reply:
@lydiasama Because you use `UpperCase` -> `TAG`, that's why I thought you intended to make it as a `const val`, but If not, can we use just a small case for `tag` 🤔  Or If you have any inside please let me know 🙏

Reply:
@Wadeewee Yeah. That's why I updated it already to be in the `companion object`. Please check again. 😬